src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,97 @@
-export const Pagination = () => {};
+import { getNumbers, getPaginatedItems } from '../../utils';
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  setCurrentPage: (arg: number) => void;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage = 1,
+  setCurrentPage,
+}: Props) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+  const items = getNumbers(1, total).map(n => `Item ${n}`);
+  const paginatedItems = getPaginatedItems(items, perPage);
+
+  const isCurrentPageFirst = currentPage === 1;
+  const isCurrentPageLast = currentPage === pages.length;
+
+  const handleBackwardsClick = () => {
+    if (!isCurrentPageFirst) {
+      setCurrentPage(currentPage - 1);
+    }
+  };
+
+  const handleForwardsClick = () => {
+    if (!isCurrentPageLast) {
+      setCurrentPage(currentPage + 1);
+    }
+  };
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={classNames('page-item', { disabled: isCurrentPageFirst })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
```

Comment:
Using `href="#prev"` can lead to unexpected behavior when the link is clicked, as it will change the URL fragment. It is better to use `preventDefault` in the `handleBackwardsClick` to avoid this behavior.